### DISCUSSION TITLE
How to get proxy on mtproto.

### DISCUSSION THREAD

**Discussion by abserari** [xelaj/mtproto#55] [Q&A] [answered]
Find this could get proxy. https://github.com/cjongseok/mtproto

**Comment by quenbyako** [comment, marked as answer]
Currently there is no simple way to set proxy in this lib, but this feature is on the roadmap. contributions are welcome

p.s. i said about contributions cause our team don't know how to implement proxy, and honestly, we don't need it. But! I agree that mtproxy is a part of telegram api, so i'll definitely accept your contribution, and if possible help you with the implementation

**Reply by abserari** [reply]
I try to add socks5 proxy, but it seems to have a response decode error. I think there is some problem with the proxy message? In the auth example, 
```
PS D:\gopath\src\github.com\abserari\mtproto\examples\auth> go run .\main.go 17336329520
MessageContainer
session created
panic: (*serialize.BadMsgNotification) 0xc000253830 [recovered]
        panic: (*serialize.BadMsgNotification) 0xc000253830

goroutine 7 [running]:
github.com/xelaj/mtproto.(*MTProto).recoverGoroutine(0xc000203b80)
        D:/gopath/src/github.com/abserari/mtproto/mtproto_utils.go:77 +0x125
panic(0x5bf060, 0xc000253830)
        c:/go/src/runtime/panic.go:969 +0x1c7
github.com/xelaj/mtproto.(*MTProto).processResponse(0xc000203b80, 0x5ff59cd3802e8401, 0x4, 0x69f5e0, 0xc00025b260, 0xc00025b260, 0x0)
        D:/gopath/src/github.com/abserari/mtproto/mtproto.go:357 +0x892
github.com/xelaj/mtproto.(*MTProto).startReadingResponses.func1(0xc000203b80, 0x69fba0, 0xc00005fe40)        
        D:/gopath/src/github.com/abserari/mtproto/mtproto.go:284 +0x2b5
created by github.com/xelaj/mtproto.(*MTProto).startReadingResponses
        D:/gopath/src/github.com/abserari/mtproto/mtproto.go:260 +0x7b
exit status 2
```

**Reply by quenbyako** [reply]
@abserari i need to check this message, honestly, i don't know why is this happening, cause we're removed all panics in decoder.

Maybe you're using non rc branch?

Also, @abserari, this is important, i've found [this code](https://github.com/xelaj/mtproto/compare/master...abserari:master#diff-747d3ea37216dacdfd210de691ba96cddde503cd798ba35a9394b777c77f88a7R33), which contains login and password to your proxy. Be sure that this is not a secret info, if it is, delete this commit as fast as you can.

**Comment by passerbyo** [comment]
i suggest add this project https://github.com/ginuerzh/gost to mtproto

**Reply by quenbyako** [reply]
How good is it? Can we add this as a good way to make proxy? Can we translate readme page of this project?
